src/modules/Dashboard/features/Applet/Overview/Overview.utils.tsx

```diff
@@ -0,0 +1,151 @@
+      },
+    ],
+  };
+}
+
+export function mapResponseToSubmissionsTableProps({
+  submissions = [],
+  submissionsCount = 0,
+}: GetAppletSubmissionsResponse) {
+  return {
+    'data-testid': 'recent-submissions',
+    columns: [
+      { id: 'activity', label: i18n.t('appletOverview.columnActivity') },
+      { id: 'respondent', label: i18n.t('appletOverview.columnRespondent') },
+      { id: 'subject', label: i18n.t('appletOverview.columnSubject') },
+      { id: 'submissionDate', label: i18n.t('appletOverview.columnSubmissionDate') },
+    ],
+    count: submissionsCount ?? submissions?.length,
+    handleRequestSort: () => {},
+    order: 'desc' as const,
+    orderBy: '',
+    rows:
+      submissions.length > 0
+        ? submissions.map(
+            ({
+              activityName,
+              createdAt,
+              sourceSubjectId,
+              sourceNickname,
+              sourceSubjectTag,
+              targetSubjectId,
+              targetNickname,
+              targetSubjectTag,
+            }) => {
+              const createdAtDate = new Date(createdAt);
+
+              return {
+                activity: {
+                  content: () => <StyledMaybeEmpty>{activityName}</StyledMaybeEmpty>,
+                  value: true,
+                },
+                respondent: {
+                  content: () => (
+                    <StyledMaybeEmpty>
+                      <ParticipantSnippet
+                        tag={sourceSubjectTag}
+                        secretId={sourceSubjectId}
+                        nickname={sourceNickname}
+                      />
+                    </StyledMaybeEmpty>
+                  ),
+                  sx: { maxWidth: theme.spacing(31.2) },
+                  value: true,
+                },
+                subject: {
+                  content: () => (
+                    <StyledMaybeEmpty>
+                      {sourceSubjectId === targetSubjectId ? (
+                        i18n.t('appletOverview.labelSelf')
+                      ) : (
+                        <ParticipantSnippet
+                          tag={targetSubjectTag}
+                          secretId={targetSubjectId}
+                          nickname={targetNickname}
+                        />
+                      )}
+                    </StyledMaybeEmpty>
+                  ),
+                  sx: { maxWidth: theme.spacing(31.2) },
+                  value: true,
+                },
+                submissionDate: {
+                  content: () => (
+                    <StyledMaybeEmpty>
+                      {formatDistanceStrict(
+                        new Date(
+                          createdAtDate.getTime() - createdAtDate.getTimezoneOffset() * 60000,
+                        ),
+                        new Date(),
+                        {
+                          addSuffix: true,
+                          locale: locales[i18n.language as keyof typeof locales],
```

Comment:
You could also use the `Languages` const in `api.const.ts`:
```suggestion
                          locale: locales[i18n.language as Languages],
```